CMakeLists.txt

```diff
@@ -50,7 +50,17 @@ configure_file(
   ${CMAKE_BINARY_DIR}/git_revision.h
 )
 
-set(FLAGS_COMMON "-msse2 -D__MMX__ -D__SSE__ -D__SSE2__ -D__USE_LARGEFILE64 -pthread")
+set(FLAGS_COMMON "-D__USE_LARGEFILE64 -pthread")
+set(supports32bit true)
+set(x86ish false)
```

Comment:
It seems kind of weird to have these start in opposite positions and then have one branch of the if flip each. Maybe just start with the x86 values and have the aarch64 branch clip both.

Reply:
The idea here is that eventually aarch64 may support the 32 bit variant also, so true is a better default, but neither it nor any future additions will ever be x86ish

Reply:
Alright fair.